Core/Src/circular_buffer.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * circular_buffer.c
+ *
+ *  Created on: Oct 1, 2024
+ *      Author: diego
+ */
+
+#include "circular_buffer.h"
+
+bool circ_buf_full(circ_buf_t *p_circ_bbuf_t, uint8_t maxlen) {
+    return (p_circ_bbuf_t->num_entries == maxlen);
+}
+
+bool circ_buf_empty(circ_buf_t *p_circ_bbuf_t) {
+    return (p_circ_bbuf_t->num_entries == 0);
+}
+
+int circ_buf_enqueue(circ_buf_t *pBuffer, float data, uint8_t maxlen) {
```

Comment:
Since these functions are exposed in `circular_buffer.h` make sure you follow the coding standard for public functions. See the Coding Style here: https://wiki.ubcsolar.com/subteams/embedded/coding_style
 
Of course this is still a solar sandbox project, however, to make your refactor and integration to production code later, I recommend following the coding style from the beginning.

Reply:
Unclear on what coding standard isn't being followed?